diff --git a/node_modules/expo-av/android/CMakeLists.txt b/node_modules/expo-av/android/CMakeLists.txt
index e55dbb2..42d7eff 100644
--- a/node_modules/expo-av/android/CMakeLists.txt
+++ b/node_modules/expo-av/android/CMakeLists.txt
@@ -34,6 +34,14 @@ find_library(LOG_LIB log)
 
 find_package(ReactAndroid REQUIRED CONFIG)
 
+if (TARGET ReactAndroid::reactnativejni)
+        set(REACT_ANDROID_JNI_TARGET ReactAndroid::reactnativejni)
+elseif (TARGET ReactAndroid::reactnative)
+        set(REACT_ANDROID_JNI_TARGET ReactAndroid::reactnative)
+else ()
+        message(FATAL_ERROR "expo-av: ReactAndroid JNI target not found")
+endif ()
+
 find_package(fbjni REQUIRED CONFIG)
 
 # linking
@@ -43,6 +51,6 @@ target_link_libraries(
         ${LOG_LIB}
         fbjni::fbjni
         ReactAndroid::jsi
-        ReactAndroid::reactnativejni
+        ${REACT_ANDROID_JNI_TARGET}
         android
 )
diff --git a/node_modules/expo-av/android/src/main/java/expo/modules/av/player/SimpleExoPlayerData.java b/node_modules/expo-av/android/src/main/java/expo/modules/av/player/SimpleExoPlayerData.java
index 672871d..de4c192 100644
--- a/node_modules/expo-av/android/src/main/java/expo/modules/av/player/SimpleExoPlayerData.java
+++ b/node_modules/expo-av/android/src/main/java/expo/modules/av/player/SimpleExoPlayerData.java
@@ -13,6 +13,7 @@ import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
 import com.google.android.exoplayer2.C;
+import com.google.android.exoplayer2.DefaultRenderersFactory;
 import com.google.android.exoplayer2.MediaItem;
 import com.google.android.exoplayer2.PlaybackException;
 import com.google.android.exoplayer2.PlaybackParameters;
@@ -86,10 +87,13 @@ class SimpleExoPlayerData extends PlayerData
     final TrackSelector trackSelector = new DefaultTrackSelector(context, new AdaptiveTrackSelection.Factory());
 
     // Create the player
-    mSimpleExoPlayer = new SimpleExoPlayer.Builder(context)
-        .setTrackSelector(trackSelector)
-        .setBandwidthMeter(bandwidthMeter)
-        .build();
+    DefaultRenderersFactory renderersFactory = new DefaultRenderersFactory(context);
+    renderersFactory.setEnableDecoderFallback(true);
+
+    mSimpleExoPlayer = new SimpleExoPlayer.Builder(context, renderersFactory)
+      .setTrackSelector(trackSelector)
+      .setBandwidthMeter(bandwidthMeter)
+      .build();
 
     mSimpleExoPlayer.addListener(this);
 
